src/rules/noBooleanLiteralCompareRule.ts

```diff
@@ -62,17 +62,32 @@ class Walker extends Lint.ProgramAwareRuleWalker {
         }
 
         const deleted = node.left === expression
-            ? Lint.Replacement.deleteFromTo(node.left.end, node.end)
-            : Lint.Replacement.deleteFromTo(node.getStart(), node.right.getStart());
+            ? this.deleteFromTo(node.left.end, node.end)
+            : this.deleteFromTo(node.getStart(), node.right.getStart());
         const replacements = [deleted];
         if (negate) {
-            replacements.push(this.appendText(node.getStart(), "!"));
+            if (needsParenthesesForNegate(expression)) {
+                replacements.push(this.appendText(node.getStart(), "!("));
+                replacements.push(this.appendText(node.getEnd(), ")"));
+            } else {
+                replacements.push(this.appendText(node.getStart(), "!"));
+            }
         }
 
         this.addFailureAtNode(expression, Rule.FAILURE_STRING(negate), this.createFix(...replacements));
     }
 }
 
+function needsParenthesesForNegate(node: ts.Expression) {
+    switch (node.kind) {
+        case ts.SyntaxKind.AsExpression:
+        case ts.SyntaxKind.BinaryExpression:
```

Comment:
I can't think of a better way to do this, unfortunately.
A ternary operator wouldn't happen here since it `===` binds more tightly, so the ternary would have to be parenthesized already anyway.